File: src/main/java/com/example/usercrud/database/adapter/JDBCConnectionConfig.java
package com.example.usercrud.database.adapter;

import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

@Configuration
@Slf4j
public class JDBCConnectionConfig {
    @Bean
    public Connection getConnection(@Value("${jdbc.driverName}") String jdbcDriverName, @Value("${jdbc.connectionURL}")String jdbcConnectionURL) throws SQLException, ClassNotFoundException {
        try {
            Class.forName(jdbcDriverName);
            return DriverManager.getConnection(jdbcConnectionURL, "admin", "password");
        } catch (Exception e) {
            e.printStackTrace();
            throw e;
        }
    }
}

File: src/main/kotlin/cloud/folium/usercrud/database/adapter/JDBCConnectionConfig.kt
package cloud.folium.usercrud.database.adapter

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.lang.Exception
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException
import kotlin.Throws

@Configuration
class JDBCConnectionConfig {
    @Bean
    @Throws(SQLException::class, ClassNotFoundException::class)
    fun getConnection(
        /*@Value("\${jdbc.driverName}") jdbcDriverName: String?,
        @Value("\${jdbc.connectionURL}") jdbcConnectionURL: String?

         */
    ): Connection {
        return try {
            val jdbcDriverName: String = "org.h2.Driver"
            val jdbcConnectionURL: String = "jdbc:h2:file:./src/main/java/com/example/usercrud/Database/Database"
            Class.forName(jdbcDriverName)
            DriverManager.getConnection(jdbcConnectionURL, "admin", "password")
        } catch (e: Exception) {
            e.printStackTrace()
            throw e
        }
    }
}